### ISSUE TITLE
Can Windows arm64 be supported

### ISSUE THREAD

**Issue by cheezmil** [sabrogden/Ditto#654]
I have a Windows tablet with a Snapdragon 8cx gen3 CPU. As the title suggests, thank you

**Comment by tg2k** [comment]
It should be possible to adjust x64 checks since ARM64 is compatible with x64 through emulation, but it looks like genuine ARM64 work was done in https://github.com/sabrogden/Ditto/issues/362 and https://github.com/sabrogden/Ditto/pull/376 . Not sure if @jglathe or @sabrogden know more about what else is needed to release an ARM64 install?

**Comment by jglathe** [comment]
Hi, I'm using it on arm64 for a long time now. Repo is [here](https://github.com/jglathe/Ditto), I also have made a release package. I could make an updated one if required, but no (new) problems arised.

**Comment by jglathe** [comment]
https://github.com/jglathe/Ditto/releases/tag/3.24.246.1

**Comment by tg2k** [comment]
> https://github.com/jglathe/Ditto/releases/tag/3.24.246.1

I checked this out shortly after you posted it, and was surprised to find that your build is larger than the prior x64 build. 6213KB vs 5337KB. Generally I hear that ARM64 results in smaller binaries. Do you have any idea why?

**Comment by jglathe** [comment]
Actually no idea. Never checked. I have some more commits in it, but I left out the new sqlite version (that one gave lots of compile errors). Working flawlessly here, though.

**Comment by aplotor** [comment]
hey thanks for doing that @jglathe

i'm thinking of getting an arm laptop, can i ask you some questions?

1. does the official (non-arm) release work on arm and if so how's the performance?
2. does your arm release support ditto portable mode?

thank you

**Comment by tg2k** [comment]
@sabrogden I tried installing v3.25.76 via Chocolatey on ARM64 to see if it had been made compatible, and it looks like not. In case it helps, here is a portion of the Chocolatey log:

`2025-09-06 10:00:17,664 22420 [INFO ] - Installing ditto...
2025-09-06 10:00:17,749 22420 [DEBUG] - Running Start-ChocolateyProcessAsAdmin -validExitCodes '0' -workingDirectory 'C:\ProgramData\chocolatey\lib\ditto\tools' -statements '/VERYSILENT /SUPPRESSMSGBOXES /NORESTART /CLOSEAPPLICATIONS /SP-' -exeToRun 'C:\ProgramData\chocolatey\lib\ditto\tools\DittoSetup_3_25_76_0.exe' 
2025-09-06 10:00:17,754 22420 [DEBUG] - Running Test-ProcessAdminRights
2025-09-06 10:00:17,756 22420 [DEBUG] - Test-ProcessAdminRights: returning True
2025-09-06 10:00:17,757 22420 [DEBUG] - Finishing 'Test-ProcessAdminRights'
2025-09-06 10:00:17,773 22420 [DEBUG] - Elevating permissions and running ["C:\ProgramData\chocolatey\lib\ditto\tools\DittoSetup_3_25_76_0.exe" /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /CLOSEAPPLICATIONS /SP-]. This may take a while, depending on the statements.
2025-09-06 10:00:18,813 22420 [DEBUG] - Command ["C:\ProgramData\chocolatey\lib\ditto\tools\DittoSetup_3_25_76_0.exe" /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /CLOSEAPPLICATIONS /SP-] exited with '1'.
2025-09-06 10:00:18,862 22420 [ERROR] - ERROR: Running ["C:\ProgramData\chocolatey\lib\ditto\tools\DittoSetup_3_25_76_0.exe" /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /CLOSEAPPLICATIONS /SP-] was not successful. Exit code was '1'. See log for possible error messages.
2025-09-06 10:00:18,871 22420 [DEBUG] - Built-in PowerShell host called with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = '';[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::SystemDefault; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\ditto\tools\chocolateyInstall.ps1' -installArguments '' -packageParameters '' -preRunHookScripts $null -postRunHookScripts $null'] exited with '1'.
2025-09-06 10:00:18,875 22420 [DEBUG] - Calling command ['"C:\Windows\System32\shutdown.exe" /a']
2025-09-06 10:00:18,912 22420 [DEBUG] - Command ['"C:\Windows\System32\shutdown.exe" /a'] exited with '1116'
`

**Comment by jglathe** [comment]
The normal Install works, curious?

**Comment by tg2k** [comment]
No, it gives the error: This program does not support the version of Windows your computer is running. Note that I never installed your forked version, @jglathe .

**Comment by jglathe** [comment]
Hmm. I'm not that much on Win11 arm, but I can take a look tomorrow and build the current version. As I already wrote, it was sort of fire and forget, working every time I need it and I'm on Win11 arm.

**Comment by Chealer** [comment]
Thank you @tg2k
Does this persist with 03.25.113 or ulterior? If so, please provide a new log extract which preserves formatting (line breaks).

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by tg2k** [comment]
> Thank you [@tg2k](https://github.com/tg2k) Does this persist with 03.25.113 or ulterior? If so, please provide a new log extract which preserves formatting (line breaks).

Thanks for the new version. I installed it via Chocolatey some time back and it installs and works fine.

**Comment by Chealer** [comment]
@cheezmil: Ditto is supposed to support ARM. Unless this ticket intends to report an issue, please discard it and use [the forums](https://sourceforge.net/p/ditto-cp/discussion/) instead.

**Comment by tg2k** [comment]
> [@cheezmil](https://github.com/cheezmil): Ditto is supposed to support ARM. Unless this ticket intends to report an issue, please discard it and use [the forums](https://sourceforge.net/p/ditto-cp/discussion/) instead.

It did report an issue, which is now fixed. Isn't the proper course of action for @sabrogden or @cheezmil to close it?

**Comment by Chealer** [comment]
@tg2k: This ticket does not report a bug, but it may have been meant as an implicit report that ARM was unsupported, an issue which has been solved.
In any case, someone with the necessary permissions (which does include these at least) should mark it as discarded or resolved.
